Library/RewardStateType.swift

```diff
@@ -0,0 +1,61 @@
+import KsApi
+import UIKit
+
+public enum ProjectState {
+  case nonlive
+  case live
+}
+
+public enum RewardCellProjectBackingState: Equatable {
+  case backedError(rewardState: RewardState)
+  case nonBacked(live: ProjectState, rewardState: RewardState)
+  case backed(live: ProjectState, rewardState: RewardState)
+
+  static func state(with project: Project, reward: Reward) -> RewardCellProjectBackingState {
+    let backing = project.personalization.backing
+
+    let isBacking = userIsBacking(reward: reward, inProject: project)
+
+    guard let projectBacking = backing, isBacking else {
+      return project.state == .live
+        ? nonBacked(live: .live, rewardState: RewardState.state(with: reward, project: project))
+        : nonBacked(live: .nonlive, rewardState: RewardState.state(with: reward, project: project))
+    }
+
+    switch (project.state, projectBacking.status) {
+    case(.live, .errored):
+      return .backedError(rewardState: RewardState.state(with: reward, project: project))
+    case(.live, _):
+      return .backed(live: .live, rewardState: RewardState.state(with: reward, project: project))
+    case (_, _):
+      return .backed(live: .nonlive, rewardState: RewardState.state(with: reward, project: project))
+    }
+  }
+
+  public enum RewardState {
+    case limited
```

Comment:
What would the `RewardState` be for a reward that is neither limited, nor time based? Can we add a test to see what the state function would return with a reward of type `Reward.noReward`?

Reply:
A reward that is neither limited, nor time based  would be `inactive`